assets/src/components/detourListPage.tsx

```diff
@@ -29,19 +60,35 @@ export const DetourListPage = () => {
       )}
       {detours && (
         <>
-          {detours.active && <DetoursTable data={detours.active} />}
-          {detours.draft && <DetoursTable data={detours.draft} />}
-          {detours.past && <DetoursTable data={detours.past} />}
+          {detours.active && (
+            <DetoursTable data={detours.active} onOpenDetour={onOpenDetour} />
+          )}
+          {detours.draft && (
+            <DetoursTable data={detours.draft} onOpenDetour={onOpenDetour} />
+          )}
+          {detours.past && (
+            <DetoursTable data={detours.past} onOpenDetour={onOpenDetour} />
+          )}
         </>
       )}
 
-      {showDetourModal && (
+      {/* Need to be separate modals to force the modal's state machine to refresh */}
```

Comment:
```suggestion
      {/* React's state caching requires that separate modals are used to force the state machine to refresh. the `key` prop didn't reset the state either when tried with `key={detourId ?? ""}` */}
```

Reply:
Ok so, I went the two-modal method for a different reason, that I just realized I was thinking about it the wrong way. The `key` prop does reset the state, and now that I adjusted the `DetourModal` conditional better, things render well. 

I needed to delay showing the modal between the time where `detourId` exists but `stateOfDetourModal` doesn't exist yet. Updated the comment to reflect that!

Reply:
Oh interesting! I was testing locally and didn't think about that and was having issues!